wsl-pro-service/internal/daemon/daemon_test.go

```diff
@@ -367,19 +367,22 @@ func TestRetryLogic(t *testing.T) {
 			ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
 			if tc.precancelled {
 				cancel()
+				<-ctx.Done() // Otherwise the test may finish before the context is trully cancelled.
 			} else {
 				defer cancel()
 			}
-			retryCalled := false
+			tryCount := 0
 			tooManyAttempts := false
 			rc := daemon.NewRetryConfig(minWait, maxWait, maxRetries)
+			// All functions passed below run in the same goroutine, thus no need for
+			// synchronisation.
 			err := rc.Run(ctx, func() (bool, error) {
 				return tc.succeedWithoutRetries, tc.actionError
 			}, func(wait time.Duration) {
 				if tc.wantNoRetries {
-					require.Fail(t, "Unexpected Retry attempt")
+					require.LessOrEqual(t, tryCount, 1, "Unexpected Retry attempt")
```

Comment:
Should this be `require.Less` (or just assert 0) instead of `require.LessOrEqual`? For example, if we get here after the first attempt, `tryCount` is 1, but it should fail here since we don't want to have called the wait function again.